driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocument.java

```diff
@@ -430,6 +471,21 @@ public BsonDateTime getWallTime() {
         return wallTime;
     }
 
+
+    /**
+     * When the change stream's backing aggregation pipeline contains the
+     * <code>$changeStreamSplitLargeEvent</code> stage, events larger than 16MB
+     * will be split into multiple events. This split event information takes
+     * the form: <code>{ fragment: &lt;current>, of: &lt;total> }</code>.
+     *
+     * @return the split event
+     * @since 4.11
+     * @mongodb.server.release 7.0
+     */
+    public BsonDocument getSplitEvent() {
+        return splitEvent;
```

Comment:
This could return a proper class, similar to `UpdateDescription`, but it produces a `BsonDocument`, like `lsid`. I do not see a clear way to decide this. I chose this because it seemed uncommon.

Reply:
`lsid` is not supposed to be deconstructed by a user, while deconstructing `splitEvent` is pretty much a requirement if a user wants to meaningfully use change events that are split into fragments (though, I have no idea if the MongoDB Kafka Connector will have any automatic handling for those in the future, and I understand even less about the MongoDB Spark Connector). It seems to me that giving users a convenient type to work with makes all the sense here.